esmvaltool/diag_scripts/mlr/custom_sklearn.py

```diff
@@ -0,0 +1,771 @@
+"""Custom expansions of :mod:`sklearn` functionalities."""
+
+# pylint: disable=arguments-differ
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=protected-access
+# pylint: disable=super-init-not-called
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-locals
+
+import itertools
+import logging
+import numbers
+import os
+import time
+import warnings
+from contextlib import suppress
+from copy import deepcopy
+from inspect import getfullargspec
+from traceback import format_exception_only
+
+import numpy as np
+from joblib import Parallel, delayed, effective_n_jobs
+from sklearn.base import BaseEstimator, clone, is_classifier
+from sklearn.compose import ColumnTransformer, TransformedTargetRegressor
+from sklearn.exceptions import FitFailedWarning, NotFittedError
+from sklearn.feature_selection import RFE
+from sklearn.feature_selection._base import SelectorMixin
+from sklearn.linear_model import LinearRegression
+from sklearn.metrics import check_scoring
+from sklearn.metrics._scorer import (
+    _check_multimetric_scoring,
+    _MultimetricScorer,
+)
+from sklearn.model_selection import check_cv
+from sklearn.model_selection._validation import _aggregate_score_dicts, _score
+from sklearn.pipeline import Pipeline
+from sklearn.utils import (
+    _message_with_time,
+    check_array,
+    check_X_y,
+    indexable,
+    safe_sqr,
+)
+from sklearn.utils.metaestimators import _safe_split, if_delegate_has_method
+from sklearn.utils.validation import _check_fit_params, check_is_fitted
+
+from esmvaltool.diag_scripts import mlr
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def _fit_and_score_weighted(estimator, x_data, y_data, scorer, train, test,
+                            verbose, parameters, fit_params,
+                            error_score=np.nan, sample_weights=None):
+    """Expand :func:`sklearn.model_selection._validation._fit_and_score`."""
+    if verbose > 1:
+        if parameters is None:
+            msg = ''
+        else:
+            msg = '%s' % (', '.join('%s=%s' % (k, v)
+                                    for k, v in parameters.items()))
+        print("[CV] %s %s" % (msg, (64 - len(msg)) * '.'))
```

Comment:
logger instead of print?

Reply:
I want to stay as close to the original `sklearn` implementation as possible, so I'd like to keep that.

Reply:
ok